### PR TITLE
Add groupby `max` aggregation benchmark

### PR DESCRIPTION
**Pull request by ttnghia**
This adds a simple benchmark for groupby `max` aggregation.

### CODE REVIEW

**Review by karthikeyann — APPROVED** [review]
LGTM 👍

File: cpp/benchmarks/groupby/group_max.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+#include <nvbench/nvbench.cuh>
+
+template <typename Type>
+void bench_groupby_max(nvbench::state& state, nvbench::type_list<Type>)
+{
+  cudf::rmm_pool_raii pool_raii;
+
+  auto const input_table = [&] {
+    data_profile profile;
+    profile.set_null_frequency(std::nullopt);
```
Comment (davidwendt) [inline cpp/benchmarks/groupby/group_max.cpp:31]: I would be curious to know the performance impact when including nulls.
Could this be added as a state parameter? 
Perhaps just a single value like 10% or less would be enough.

Comment (ttnghia) [reply cpp/benchmarks/groupby/group_max.cpp:32]: I've added nulls for the values column.

Comment (davidwendt) [reply cpp/benchmarks/groupby/group_max.cpp:32]: Ok, that makes sense.

**Review by davidwendt — APPROVED** [review]

**Comment by ttnghia** [comment]
@gpucibot merge

**Pull request merged**